src/main/java/smartsnake/astar/kenigsberg/AstarBrain.java

```diff
@@ -0,0 +1,75 @@
+package smartsnake.astar.kenigsberg;
+
+import smartsnake.*;
+import smartsnake.astar.Node;
+
+import java.awt.*;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AstarBrain implements Brain {
+
+    @Override
+    public Direction move(Snake snake, Food food, Garden garden) {
+
+        ArrayList<Node> openNodes = new ArrayList<>();
+        ArrayList<Node> closedNodes = new ArrayList<>();
+
+        Node startNode = new Node(snake.getHeadLocation(), food.getLocation());
+        openNodes.add(startNode);
+
+        addSnakeToClosed(snake, food, closedNodes);
+
+        while (!openNodes.isEmpty()) {
+            Node current = getLowestCost(openNodes);
+
+            openNodes.remove(current);
+            closedNodes.add(current);
+
+            Point currentLocation = current.getLocation();
+            if (currentLocation.equals(food)) {
+                return current.getDirectionFromStart();
+            }
+
+            updateNeighbors(current, new Point(currentLocation.x + 1, currentLocation.y),
+                    Direction.Right, openNodes, closedNodes, food);
+            updateNeighbors(current, new Point(currentLocation.x, currentLocation.y - 1),
+                    Direction.Up, openNodes, closedNodes, food);
+            updateNeighbors(current, new Point(current.getLocation().x - 1, current.getLocation().y),
+                    Direction.Left, openNodes, closedNodes, food);
+            updateNeighbors(current, new Point(currentLocation.x, currentLocation.y + 1),
+                    Direction.Down, openNodes, closedNodes, food);
+        }
+        return null;
+    }
+
+    public void updateNeighbors(Node current, Point point, Direction direction,
+                                ArrayList<Node> openNodes, ArrayList<Node> closedNodes, Food food) {
+
+        Node directionNode = new Node(point, food);
+        if (!closedNodes.contains(directionNode)) {
+            if (!openNodes.contains(directionNode)) {
+                openNodes.add(directionNode);
+            }
+            openNodes.get(openNodes.indexOf(directionNode)).setParent(current, direction);
```

Comment:
This is what I was getting at.

```suggestion
            int index = openNodes.indexOf(directionNode)
            if (index == -1) {
                openNodes.add(directionNode);
            }
            else {
                directionNode = openNodes.get(index);
            }
            directionNode.setParent(current, direction);
```